client/src/components/History/ContentItem/ContentItem.js

```diff
@@ -1,83 +0,0 @@
-/**
```

Comment:
Not entirely with you on this one, I'd prefer a separation in the Vue-template if necessary, personally I found these abstraction layers cumbersome and difficult to follow.

Reply:
So currently the `ContentItem` uses `CollectionDescription` and `DatasetDetails` as sub-components to build the collapsed history items and collection elements see: 
<img width="282" alt="image" src="https://user-images.githubusercontent.com/2105447/157152158-1740ada3-0d28-44b4-9ad2-ba88516ab9f5.png">

Reply:
I looked more into this, and I see that there were previously 30 files to render the history items and now there are just 5 and they are much smaller. I think the reason is that generic shim components like above can be good if used in moderation but also can produce significant redundancies, overhead since they do not contain a lot information by themselves. Additionally I think that they can make development and debugging particularly for new and changing project much more challenging .

Reply:
I prefer the extra separation and abstraction only when it's actually useful to reuse as a component elsewhere (and when it's actually used -- i.e. build first and factor out reusable chunks later if you need it), or when it boxes up complicated functionality in a way that truly makes it easier to use and reason about.

A lot of the previous code split things apart just because it could (the components with single line state descriptions were overkill), and I think the right spot for us is much closer to where @guerler is headed.

Reply:
> I prefer the extra separation and abstraction only when it's actually useful to reuse as a component elsewhere (and when it's actually used -- i.e. build first and factor out reusable chunks later if you need it), or when it boxes up complicated functionality in a way that truly makes it easier to use and reason about.

I totally agree, that's why I commented on this component, and not others that are overly complex. Look at the if/else statements in ContentItem.vue now. I don't think this is better, but I can appreciate making progress over a cleaner separation.